emoji/src/main/java/com/gluonhq/emoji/DownloadableEmojiSpriteLoader.java

```diff
@@ -0,0 +1,176 @@
+            } catch (Exception e) {
+                LOG.log(Level.SEVERE, "Downloading of sprites failed", e);
+                return false;
+            }
+        }, Executors.newSingleThreadExecutor(r -> {
+            Thread t = Executors.defaultThreadFactory().newThread(r);
+            t.setDaemon(true);
+            return t;
+        }));
+    }
+
+    private void downloadSprites(int... sizes) {
+        for (int size : sizes) {
+            if (!localFileExists(size)) {
+                try {
+                    Path localFilePath = getLocalFilePath(size);
+                    Path parentDir = localFilePath.getParent();
+                    if (parentDir != null && !Files.exists(parentDir)) {
+                        Files.createDirectories(parentDir);
+                    }
+                    Path tmpFilePath = getTmpFilePath(size);
+                    LOG.fine("Creating tmp file: " + tmpFilePath);
+                    Files.createFile(tmpFilePath);
+                    LOG.fine("Download sprite file for size: " + size);
+                    String url = String.format(EMOJI_PNG_URL, commit, size);
+                    downloadFile(new URI(url).toURL(), localFilePath, tmpFilePath);
+                } catch (IOException | URISyntaxException e) {
+                    LOG.severe("Download sprite failed: " + e.getMessage());
+                    throw new RuntimeException("Unable to load local image file", e);
+                }
+            }
+        }
+    }
+
+    public Image loadEmojiSprite(int size) {
+        if (!initialized) {
+            throw new RuntimeException("Sprite Loader hasn't been initialized or completed initialization");
+        }
+        try {
+            return new Image(new FileInputStream(getLocalFilePath(size).toFile()));
+        } catch (IOException e) {
+            LOG.severe("Loading of local image file failed: " + e.getMessage());
+            throw new RuntimeException("Unable to load local image file", e);
+        }
+    }
+
+    @Override
+    public InputStream loadCSV() {
+        return DownloadableEmojiSpriteLoader.class.getResourceAsStream("emoji.csv");
+    }
+
+    private void downloadFile(URL url, Path filePath, Path tmpFile) throws IOException {
+        try (InputStream inputStream = url.openStream()) {
+            Files.copy(inputStream, filePath, StandardCopyOption.REPLACE_EXISTING);
+            LOG.fine("Sprite downloaded successfully as: " + filePath);
+            LOG.fine("Removing tmp file: " + tmpFile);
+            Files.delete(tmpFile);
+        } catch (IOException e) {
+            LOG.log(Level.SEVERE, "Downloading file failed", e);
+        }
+    }
+
+    private boolean localFileExists(int size) {
+        Path localFilePath = getLocalFilePath(size);
+        Path tmpFilePath = getTmpFilePath(size);
+        if (Files.exists(tmpFilePath)) {
+            LOG.fine("Temporary download file found");
+            LOG.fine("Delete file and re-try downloading");
```

Comment:
Do you want all logging to happen at `info` level or just this one?

Reply:
Just that one. 
While we are developing ourselves, we can easily increase the log level to check detailed logging.
But in case this is happening with users, and when they submit logs, only logs with INFO or higher are considered.